aws/resource_aws_fms_admin_account.go

```diff
@@ -0,0 +1,125 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/fms"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsFmsAdminAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsFmsAdminAccountPut,
+		Read:   resourceAwsFmsAdminAccountRead,
+		Delete: resourceAwsFmsAdminAccountDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(1 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsAccountId,
+			},
+		},
+	}
+}
+
+func resourceAwsFmsAdminAccountPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).fmsconn
+
+	accountId := d.Get("account_id").(string)
+
+	stateConf := &resource.StateChangeConf{
+		Target:     []string{accountId},
+		Refresh:    associateAdminAccountRefreshFunc(conn, accountId),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for firewall manager admin account association: %v", accountId)
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for firewall manager admin account association (%s): %s", accountId, sterr)
+	}
+
+	d.SetId("fms-admin-account")
```

Comment:
While we don't necessarily need to reference it during read/delete at the moment, can you set this to `accountId` still? It'll make the ID available if we need it in the future and the import a little more obvious:

```
# Currently:
terraform import aws_fms_admin_account.example fms-admin-account
# Proposed:
terraform import aws_fms_admin_account.example 123456789012
```

Reply:
@bflad I adopted this approach from `aws_iam_account_password_policy` (`aws_spot_datafeed_subscription` does this too) since only one FMS admin account can be associated to an account. My concern is that the user can disassociate/associate a different admin account outside Terraform - it might be confusing/misleading to have the resource ID as an account ID different to associated admin account ID. I'm guessing it wouldn't be safe to call `SetId` during a read operation if the associated admin account has changed? Any suggestions?